### PR TITLE
RegionValue doesn't extend Serializable

### PR DESCRIPTION
**Pull request by catoverdrive**
Currently still being KryoSerialized in a lot of places (apparently), which I'm going to deal with next. This mostly just changes how pkis are stored and passed around.

### CODE REVIEW

File: src/main/scala/is/hail/rvd/OrderedRVPartitionInfo.scala [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 case class OrderedRVPartitionInfo(
   partitionIndex: Int,
   size: Int,
-  min: RegionValue,
-  max: RegionValue,
+  min: SerializedRegionValue,
+  max: SerializedRegionValue,
   // min, max: RegionValue[pkType]
-  samples: Array[RegionValue],
+  samples: Array[SerializedRegionValue],
```
Comment (cseed) [inline src/main/scala/is/hail/rvd/OrderedRVPartitionInfo.scala]: Curious, why not just make these (safe) annotations?

**Comment by catoverdrive** [comment]
ok @patrick-schultz I finished fixing this up.

File: src/main/scala/is/hail/rvd/OrderedRVD.scala [hunk 2]
```diff
@@ -479,7 +479,8 @@ object OrderedRVD {
         Iterator()
     }.run.collect()
 
-    pkis.sortBy(_.min)(typ.pkOrd)
+    val region = Region()
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/rvd/OrderedRVD.scala]: not used

File: src/main/scala/is/hail/rvd/OrderedRVD.scala [hunk 3]
```diff
@@ -548,8 +549,10 @@ object OrderedRVD {
     if (pkis.isEmpty)
       return empty(sc, typ)
 
+    val region = Region()
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/rvd/OrderedRVD.scala]: not used

File: src/main/scala/is/hail/rvd/OrderedRVD.scala [hunk 4]
```diff
@@ -596,15 +599,16 @@ object OrderedRVD {
   def calculateKeyRanges(typ: OrderedRVDType, pkis: Array[OrderedRVPartitionInfo], nPartitions: Int): Array[Interval] = {
     assert(nPartitions > 0)
 
-    val pkOrd = typ.pkOrd
+    val pkOrd = typ.pkType.ordering.toOrdering
+    val region = Region()
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/rvd/OrderedRVD.scala]: not used

File: src/main/scala/is/hail/rvd/OrderedRVD.scala [hunk 5]
```diff
@@ -635,14 +639,16 @@ object OrderedRVD {
...
-    val pkOrdUnsafe = pkType.unsafeOrdering(true)
+    val pkOrd = pkType.ordering.toOrdering
     val pkis = getPartitionKeyInfo(typ, OrderedRVD.getKeys(typ, rdd))
 
     if (pkis.isEmpty)
       return OrderedRVD(typ, partitioner, rdd)
 
-    val min = new UnsafeRow(pkType, pkis.map(_.min).min(pkOrdUnsafe))
-    val max = new UnsafeRow(pkType, pkis.map(_.max).max(pkOrdUnsafe))
+    val region = Region()
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/rvd/OrderedRVD.scala]: not used

File: src/main/scala/is/hail/rvd/OrderedRVD.scala [hunk 6]
```diff
@@ -689,14 +695,15 @@ object OrderedRVD {
...
-    val partitionBounds = new ArrayBuilder[RegionValue]()
+    val partitionBounds = new ArrayBuilder[Any]()
     val adjustmentsBuffer = new mutable.ArrayBuffer[Array[Adjustment[RegionValue]]]
     val indicesBuilder = new ArrayBuilder[Int]()
 
     var anyOverlaps = false
 
     val it = sortedKeyInfo.indices.iterator.buffered
 
+    val region = Region()
```
Comment (patrick-schultz) [inline src/main/scala/is/hail/rvd/OrderedRVD.scala]: not used

**Pull request merged**